plugins/homepagecanonical/class.homepagecanonical.plugin.php

```diff
@@ -0,0 +1,32 @@
+<?php
+/**
+ * Homepage Canonical Plugin.
+ *
+ * @author Patrick Kelly <patrick.k@vanillaforums.com>
+ * @copyright 2009-2018 Vanilla Forums Inc.
+ * @license GNU GPLv2 http://www.opensource.org/licenses/gpl-2.0.php
+ */
+
+class HomepageCanonicalPlugin extends Gdn_Plugin {
+
+    /**
+     * If the Default Controller page is requested (Discussions or Categories), 301 redirect to "/" and insert the Canonical tag.
+     *
+     * @param VanillaController $sender
+     * @param array $args
+     */
+    public function base_render_before($sender, $args) {
```

Comment:
I believe this hook is too late for wide adoption. Is there an earlier hook we can safely evaluate the URL to avoid needless database querying & page building before doing a redirect?

Reply:
I split it into to methods so that the redirecting happens in the dispatcher function.